internal/datacoord/clustering_compaction_manager.go

```diff
@@ -224,252 +206,310 @@ func (t *ClusteringCompactionManager) gc() error {
 	return nil
 }
 
-func (t *ClusteringCompactionManager) checkAllJobState() error {
-	jobs := t.GetAllJobs()
-	for _, job := range jobs {
-		err := t.checkJobState(job)
-		if err != nil {
-			log.Error("fail to check job state", zap.Error(err))
-			job.state = failed
-			log.Warn("mark clustering compaction job failed", zap.Int64("tiggerID", job.triggerID), zap.Int64("collectionID", job.collectionID))
-			t.saveJob(job)
+func (t *ClusteringCompactionManager) processAllTasks() error {
+	triggers := t.meta.GetClusteringCompactionTasks()
+	for triggerID, tasks := range triggers {
+		for _, task := range tasks {
+			log.Debug("task state before process", zap.Int64("planID", task.GetPlanId()), zap.String("state", task.State.String()))
+			err := t.processTask(task)
+			if err != nil {
+				log.Error("fail in process task", zap.Int64("TriggerId", task.TriggerId), zap.Int64("collectionID", task.CollectionId), zap.Int64("planID", task.PlanId), zap.Error(err))
+				task.State = datapb.CompactionTaskState_failed
+			}
+			log.Debug("task state after process", zap.Int64("planID", task.GetPlanId()), zap.String("state", task.State.String()))
+			t.saveTask(task)
 		}
+		state, executingCnt, pipeliningCnt, completedCnt, failedCnt, timeoutCnt, initCnt, analyzingCnt, analyzedCnt, indexingCnt, indexedCnt := t.summaryCompactionTaskState(tasks)
```

Comment:
There are an excessive number of return values.

Reply:
wrap it in a struct